lld/test/COFF/lto-cache-errors.ll

```diff
@@ -0,0 +1,20 @@
+; REQUIRES: x86
+;; Not supported on windows since we use permissions to deny the creation
+; UNSUPPORTED: system-windows
+
+; RUN: opt -module-hash -module-summary %s -o %t.o
+; RUN: opt -module-hash -module-summary %p/Inputs/lto-cache.ll -o %t2.o
+; RUN: rm -Rf %t.cache && mkdir %t.cache
+; RUN: chmod 444 %t.cache
+
+;; Check emit warnings when we can't create the cache dir
+; RUN: not --crash lld-link /lldltocache:%t.cache/cache /out:%t3 /entry:main %t2.o %t.o 2>&1 | FileCheck %s
```

Comment:
also the `--crash` is worrying, I'm guessing there's some `report_fatal_error` somewhere where we should be passing `false` as the second param to not treat it as a crash but a user error, and we should have a followup change to fix that?

Reply:
Yeah we did some discussion on the `report_fatal_error` above, I was under the impression that it always "crashed", but if it's just a bool to that function that's probably a good fix. I can follow up on that in a new PR if everyone agrees that this is a good way to handle that. cc @teresajohnson @dwblaikie